enhancements/single-node-production-deployment-approach.md

```diff
@@ -0,0 +1,484 @@
+these events. OpenShift core components are already resilient in this
+way.
+
+#### Cluster Machine Approver
+
+Auto-approval of certificate signing requests requires 2 sources of
+truth to avoid security attacks like
+[kubeletmein](https://github.com/openshift/machine-config-operator/issues/731). In
+single-node deployments we do not have a second source of truth, and
+need to disable the machine-approver-operator. An outside tool can be
+used to approve any certificate signing requests instead.
+
+#### Lack of high-availability
+
+In a single-node configuration OpenShift would have limited ability to
+provide resiliency features. We will need to clearly communicate these
+limitations to customers choosing to use single-node deployments.
+
+#### Failure to reboot
+
+It is possible that a single node may fail to reboot due to an issue
+with configuration, hardware, or environment. These single-node
+deployments will be managed by external tools, and as with other
+failures related to providing resilient operation, host boot failures
+will need to be detected and mitigated by those tools, rather than the
+local OpenShift instance.
+
+#### Reboots for reconfiguration
+
+Some reconfiguration operations will unavoidably cause the node to
+reboot. These tasks will need to be scheduled during maintenance
+windows.  Ideally we will be able to rely on the [graceful shutdown
+support in
+kubelet](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2000-graceful-node-shutdown)
+to stop all workloads safely as part of the reboot.
+
+## Design Details
+
+### Open Questions
+
+1. Telco workloads typically require special network setups
+   for a host to boot, including bonded interfaces, access to multiple
+   VLANs, and static IPs. How do we anticipate configuring those?
+2. How do we hand off ownership of the `etcd-quorum-guard` Deployment
+   from the `cluster-version-operator` to the `cluster-etcd-operator`?
+
+### Test Plan
+
+In order to claim full support for this configuration, we must have CI
+coverage informing the release. An end-to-end job using a single-node
+deployment and running an appropriate subset of the standard OpenShift
+tests will be created and configured to block accepting release images
+unless it passes.
+
+That end-to-end job should also be run against pull requests for the
+operators and other components that are most affected by the new
+configuration, such as the etcd and auth operators.
+
+There are 2 new CI jobs for testing the results. The
+e2e-metal-single-node-live-iso job
+(https://github.com/openshift/release/pull/14552) tests using the
+bootstrap-in-place approach described in
+https://github.com/openshift/enhancements/pull/565 on Packet and
+e2e-aws-single-node (https://github.com/openshift/release/pull/14556)
+uses AWS to test single-node deployments with a bootstrap VM.
+
+### Graduation Criteria
+
+##### Dev Preview
+
+- Ability to deploy an instance of OpenShift on a single host
```

Comment:
can we explicitly add the caveat that upgrades and HA (and its related functionality) are not supported for clarity

Reply:
Those are listed as non-goals up at the top of the doc.